backend/kesaseteli/companies/api/v1/views.py

```diff
@@ -10,18 +10,33 @@
 from companies.models import Company
 from companies.services import get_or_create_company_with_business_id
 from companies.tests.data.company_data import DUMMY_COMPANY_DATA
+from oidc.utils import get_organization_roles
 
 
 class GetCompanyView(APIView):
     """
     API View to retrieve company info using the YTJ API integration.
     """
 
+    @property
+    def organization_roles_error(self):
+        return Response(
+            "Unable to fetch organization roles from eauthorizations API",
+            status.HTTP_401_UNAUTHORIZED,
+        )
+
     @property
     def ytj_api_error(self):
         return Response(
             "YTJ API is under heavy load or no company found with the given business id",
-            status.HTTP_500_INTERNAL_SERVER_ERROR,
+            status.HTTP_404_NOT_FOUND,
+        )
+
+    @property
+    def ytj_response_error(self):
```

Comment:
These error response can be moved to the YTJ module in `shared` package in the refactor PR :) I think we can reuse this in Benefit